source/material_model/rheology/elasticity.cc

```diff
@@ -262,12 +261,10 @@ namespace aspect
               for (unsigned int j=0; j < SymmetricTensor<2,dim>::n_independent_components; ++j)
                 stress_old[SymmetricTensor<2,dim>::unrolled_to_component_indices(j)] = in.composition[i][j];
 
-              // Average viscoelastic viscosity
-              const double average_viscoelastic_viscosity = out.viscosities[i];
-
-              // Fill elastic force outputs (See equation 30 in Moresi et al., 2003, J. Comp. Phys.)
-              force_out->elastic_force[i] = -1. * ( average_viscoelastic_viscosity / calculate_elastic_viscosity(average_elastic_shear_moduli[i]) * stress_old );
-
+              // The viscoelastic strain rate is needed only when the Newton method or the DCP method is used for solving.
```

Comment:
```suggestion
              // The viscoelastic strain rate is needed only when the Newton or defect correction picard nonlinear solvers are selected.
```

Reply:
@naliboff Thanks very much for reviewing my codes!

I have tried the benchmark and the modified Newton solver performs poorly :-(. I think the problem is caused by the following two facts:

1. In the first 4 time steps the solver should converge within 2 iterations since plastic yielding has not occurred and the system is linear. However, the modified Newton solver needs a lot more iterations to converge. This is because on the RHS of the Newton system I have changed 
$$2\eta\dot{\boldsymbol{\varepsilon}} - \frac{\eta}{G\Delta t}\boldsymbol{\tau}^{\text{old}}$$ 
to
$$2\eta\dot{\boldsymbol{\varepsilon}}_{ve},$$

which looks identical. However, the viscoelastic strain rate is calculated by
$$\dot{\boldsymbol{\varepsilon}}_{ve} = \text{dev}(\dot{\boldsymbol{\varepsilon}}) +  \frac{\boldsymbol{\tau}^{\text{old}}}{2G\Delta t},$$
where deviator() is applied to the current strain rate (this is necessary, since the viscosity is calculated with the deviatoric strain rate). Consequently, the RHS is not precisely equal to the system residual. This problem has been fixed by restoring the original RHS, and the covergence rate is much better (see the log files below), but still not as fast as expected.

2. The GMG solver has been selected in this benchmark, which requires the viscosity to be averaged in each cell. However, function MaterialModel::MaterialAveraging::average() is called _AFTER_ MaterialModel::evaluate(), which means that $\eta$ is changed after $\partial\eta/\partial\dot{\boldsymbol{\varepsilon}}$ is calculated, hence the LHS can no longer represent the system tangent. Therefore, the modified Newton solver cannot be used simultaneously with GMG solver for now. To fix this problem we may have to do a lot more work (to calculate $\partial\eta/\partial\dot{\boldsymbol{\varepsilon}}$ after cell averaging).

The following log files are the results of the main branch, this PR (after fixing problem 1), and without cell averaging. The convergence rate without cell averaging is much faster than the previous two.
[log_main.txt](https://github.com/geodynamics/aspect/files/14417058/log_main.txt)
[log_pr5580.txt](https://github.com/geodynamics/aspect/files/14417062/log_pr5580.txt)
[log_no_avg.txt](https://github.com/geodynamics/aspect/files/14417067/log_no_avg.txt)